File: backend/src/main/java/com/plugga/backend/service/SideServiceImpl.java
package com.plugga.backend.service;

import com.plugga.backend.dao.SideDAO;
import com.plugga.backend.entity.Side;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;
import java.util.List;

@Service
public class SideServiceImpl implements SideService {
    private SideDAO sideDAO;

    @Autowired
    public SideServiceImpl(final SideDAO sideDAO) {
        this.sideDAO = sideDAO;
    }

    @Override
    @Transactional
    public List<Side> findAll() {
        return sideDAO.findAll();
    }

    @Override
    @Transactional
    public Side findById(final int id) {
        return sideDAO.findById(id);
    }

    @Override
    @Transactional
    public List<Side> findByCardId(final int cardId) {
        return sideDAO.findByCardId(cardId);
    }

    @Override
    @Transactional
    public void save(final Side side) {
        sideDAO.save(side);
    }

    @Override
    @Transactional
    public void deleteById(final int id) {
        sideDAO.deleteById(id);
    }
}

File: backend/src/main/kotlin/com/plugga/backend/service/SideServiceImpl.kt
package com.plugga.backend.service

import com.plugga.backend.repository.SideRepository
import com.plugga.backend.entity.Side
import java.util.Optional
import javax.transaction.Transactional
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

@Service
class SideServiceImpl @Autowired
constructor(private val sideRepository: SideRepository) : SideService {

    @Transactional
    override fun findAll(pageable: Pageable): Page<Side> {
        return sideRepository.findAll(pageable)
    }

    @Transactional
    override fun findById(id: Int): Side? {
        val queryResult: Optional<Side> = sideRepository.findById(id)
        return if (queryResult.isPresent) queryResult.get() else null
    }

    @Transactional
    override fun findByCardId(pageable: Pageable, cardId: Int): Page<Side> {
        return sideRepository.findByCardId(pageable, cardId)
    }

    @Transactional
    override fun save(side: Side) {
        sideRepository.save(side)
    }

    @Transactional
    override fun deleteById(id: Int) {
        sideRepository.deleteById(id)
    }
}
